### ISSUE TITLE
False negatives in UnreachableCode rule

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#3219]
1.
```kotlin
fun test(b: Boolean): Int {
    if (b) {
        return 1
    } else {
        return 2
    }
    return 0
}
```

2.
```kotlin
enum class E { A, B }

fun test(e: E): Int {
    when (e) {
        E.A -> return 1
        E.B -> return 2
    }
    return 0
}
```

3.
```kotlin
fun test(): Int {
    try {
        return 1
    } catch (e: Exception) {
        throw e
    }
    return 0
}
```
